From 19222a4e54cf735acf58500a0fffae76bc0d35bc Mon Sep 17 00:00:00 2001
From: Catalin Ionita <catalin.ionita@intel.com>
Date: Fri, 6 Feb 2015 12:15:53 +0200
Subject: [PATCH] REVERTME: drm: Fix null pointer dereference in DRM module

drm_unlock I/O control is causing a null pointer dereference if the
file lock has not previously initiliazed by calling drm_lock.

REVERTME: This fix is a temporary workaround, which still needs further
investigation. This should be reverted when a full solution is found.

Issue:GMINL-5388
Change-Id: Ieca6447337ad73d26647a00b42c4840a0baa645a
Signed-off-by: Catalin Ionita <catalin.ionita@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/drm_lock.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/drm_lock.c b/drivers/gpu/drm/drm_lock.c
index f645268..e16f23f 100644
--- a/drivers/gpu/drm/drm_lock.c
+++ b/drivers/gpu/drm/drm_lock.c
@@ -254,6 +254,11 @@ int drm_lock_free(struct drm_lock_data *lock_data, unsigned int context)
 	unsigned int old, new, prev;
 	volatile unsigned int *lock = &lock_data->hw_lock->lock;
 
+	if (!lock) {
+		DRM_ERROR("Trying to free uninitialized lock\n");
+		return 1;
+	}
+
 	spin_lock_bh(&lock_data->spinlock);
 	if (lock_data->kernel_waiters != 0) {
 		drm_lock_transfer(lock_data, 0);
-- 
1.9.1

